examples/cloudml-sentiment-analysis/scoring.py

```diff
@@ -0,0 +1,237 @@
+import numpy as np
+from sklearn import metrics
+from tensorflow import flags
+from tensorflow import gfile
+from tensorflow import logging
+
+from constants import constants
+
+
+FLAGS = flags.FLAGS
+flags.DEFINE_integer('size', 1000,
+                     'Number of records to consider for predictions.')
+flags.DEFINE_string('project_name', None, 'Name of GCP project.')
+flags.DEFINE_string('model_name', None, 'Name of Gooele Cloud ML Engine model.')
+flags.DEFINE_string('input_path', None, 'Directory with records to score.')
+flags.DEFINE_integer('random_seed', 1, 'Random seed for subset selection.')
+flags.DEFINE_integer('batch_size', 20, 'Batch size for prediction job.')
+flags.mark_flag_as_required('project_name')
+flags.mark_flag_as_required('input_path')
+flags.mark_flag_as_required('input_path')
+
+
+def get_prediction_input(files):
+  """Reads and concatenates text files in input directory.
+
+  Args:
+    files: List of `str`, containing absolut path to files to read.
+
+  Returns:
+    List of `str` containing independent text reviews.
+
+  Raises:
+    ValueError: If input files are empty.
+  """
+
+  instances = []
+  for path in files:
+    with gfile.GFile(path, 'r') as lines:
+      instances += [line for line in lines]
+  if not instances:
+    raise ValueError('No review found in input files.')
+  return instances
+
+
+def format_input(input_path, size):
+  """Reads input path, randomly selects a sub-sample and concatenates them.
+
+  Args:
+    input_path: `str`, directory to read files from.
+    size: `int`, number of files to read.
+
+  Returns:
+    List of `str` containing independent text reviews.
+  """
+
+  files = gfile.ListDirectory(input_path)
+  files = [path for path in files
+           if path.endswith(constants.FILE_EXTENSION)]
+  files = np.random.choice(files, size, replace=False)
+  files = [os.path.join(input_path, filename) for filename in files]
+  instances = get_prediction_input(files)
+  return instances
+
+
+def predict_json(project, model, instances, version=None):
+  """Sends json data to a CMLE deployed model for prediction.
+
+  Args:
+    project: str, project where the Cloud ML Engine Model is deployed.
+    model: str, model name.
+    instances: [Mapping[str: Any]], Keys should be the names of Tensors
+      your deployed model expects as inputs. Values should be datatypes
+      convertible to Tensors, or (potentially nested) lists of datatypes
+      convertible to tensors.
+    version: str, version of the model to target.
+
+  Returns:
+    Mapping[str: any]: dictionary of prediction results defined by the
+      model.
+
+  Raises:
+    RuntimeError: If the call to ml-engine returns and error.
+  """
+
+  # Create the ML Engine service object.
+  # To authenticate set the environment variable
+  # GOOGLE_APPLICATION_CREDENTIALS=<path_to_service_account_file>
+  service = discovery.build('ml', 'v1')
+  name = 'projects/{}/models/{}'.format(project, model)
+
+  if version is not None:
```

Comment:
This will append `/versions/` if `version` is an empty string (but not None). That is why the guide generally recommends implicit truth checks instead:

```python
if version:
  name += .....
```